### PR TITLE
Finish transaction.rb and transaction_repository along with tests

### PR DESCRIPTION
**Pull request by avjohnston**
Should be looking good! Might have to change a few things for hound, just let me know.

### CODE REVIEW

File: test/transaction_test.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  def test_it_exists_and_has_attributes
+    t = Transaction.new({
+                                  :id => 6,
+                                  :invoice_id => 8,
+                                  :credit_card_number => "4242424242424242",
+                                  :credit_card_expiration_date => "0220",
+                                  :result => "success",
+                                  :created_at => '2021-01-06 11:29:55 UTC',
+                                  :updated_at => '2021-01-06 11:29:55 UTC'
+                                  })
```
[the remark this answers is not in the document]
Comment (brisag) [reply test/transaction_test.rb:18]: double quoutes on results

File: test/transaction_test.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class TransactionTest < Minitest::Test
+  def test_it_exists_and_has_attributes
+    t = Transaction.new({
+                                  :id => 6,
+                                  :invoice_id => 8,
+                                  :credit_card_number => "4242424242424242",
+                                  :credit_card_expiration_date => "0220",
+                                  :result => "success",
+                                  :created_at => '2021-01-06 11:29:55 UTC',
+                                  :updated_at => '2021-01-06 11:29:55 UTC'
```
[the remark this answers is not in the document]
Comment (brisag) [reply test/transaction_test.rb:17]: some more double quotes

**Review by DougWelchons — COMMENTED** [review]
We should probably change the double quotes to singles for consistency

**Comment by brisag** [comment]
Good job at getting the expiration date to change!

**Review by Diana20920 — APPROVED** [review]
All these Hound comments! 
It looks really good Andrew 😎

**Comment by Diana20920** [comment]
### It's looking good team! 🙌🏼 
Wrapping that bow on iteration 3 and here we come iteration 4!

**Pull request merged**